src/algebra/category/CommRing/constructions.lean

```diff
@@ -106,7 +106,18 @@ section terminal
 def punit_is_terminal : is_terminal (CommRing.of.{u} punit) :=
 begin
   apply_with is_terminal.of_unique { instances := ff },
-  tidy
+  intros _,
```

Comment:
Is this giving a huge speedup? Otherwise I think the short proof is better. Non-terminal `tidy` is mostly not a bad thing, unless it's very slow.